### ISSUE TITLE
Unexpected output: "W: TW: TW: TW: TW: TW: TW: TW: TW: TW: TW: TW: TW: TW: TW: TW: T"

### ISSUE THREAD

**Issue by advanceddeveloper** [Automattic/node-canvas#1155]
## Issue <s>or Feature</s>

I tried to draw a string on the canvas, but I got very weird error and program terminated immediatelly.

## Steps to Reproduce

```js
'use strict'; // Optional

var ctx = new (require('canvas').Canvas)(1, 1).getContext('2d');
ctx.font = '1px arial';
ctx.fillText('\u8081', 0, 0);
```

Nothing has been drawn to canvas and the following text has been output to the console:

```console
> node index.js
W: TW: TW: TW: TW: TW: TW: TW: TW: TW: TW: TW: TW: TW: TW: TW: T
```

Whatever is going on here, I am hundred percent sure it is not working as expected.

## Your Environment
* **Version of node-canvas:** git-master, I built it from sources
* **Environment:** node ^9.11.0 on windows 10

**Comment by chearon** [comment]
It rendered the character for me on macOS... @zbjornson can you verify this on Windows?

Only thing I can think of is that we link to a really old GTK in the wiki. You could try [MSYS2](https://github.com/Automattic/node-canvas/wiki/Installation---Windows-%28with-MSYS2%29) to get the latest Pango etc.

**Comment by advanceddeveloper** [comment]
I need to correct myself: it is actually drawn properly. Character has beed drawn, but the main problem here is the nonsense message that has been logged to stderr. Does it come from node canvas or library? In any case, I don't think such error messages are helpful.

**Comment by chearon** [comment]
No, that almost certainly comes from Pango or another of the system libraries, but I didn't get that output either so I'm not sure. Right now I don't have Windows available to test it.

**Comment by zbjornson** [comment]
Reproduced. It comes `pango_layout_get_line` in this line:

https://github.com/Automattic/node-canvas/blob/6ef44c3ed2a3aaa76d9fd5935464a6fae48e651b/src/CanvasRenderingContext2d.cc#L2146

I'm not setup to build pango to pin point it.

**Comment by advanceddeveloper** [comment]
The only similar issue I've found related to cairo library (printing `W: T` stuff) is https://github.com/Kozea/cairocffi/issues/22 I'm posting the link here for reference, maybe it can be helpful.

**Comment by chearon** [comment]
I tried it on Windows with `canvas-prebuilt`, which uses the latest Pango/Cairo and I didn't see an error. Can you verify?

I think we should start telling people to use MSYS on Windows instead of that really old GTK bundle.

**Comment by advanceddeveloper** [comment]
@chearon 

I tested it with my old GTK (not sure how to install msys), and I indeed didn't get any warnings if font size is 0 or greater than 1. I only get warnings for font size in range **(0,1]**

> I do get some Cairo warnings when you set the font to 1px

As I know, browsers don't complain when you set font size to 1px, so if you succeedeed to reproduce the warning with the latest library binaries, I think this can be considered as a bug...
